Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -0,0 +1,348 @@
+     all, still fulfil safety requirements.
+
+     As mentioned in the previous note, it is unnecessary to expect that
+     the bootloader (or the hardwired pull up/down) settings will configure
+     the same period which will be enforced past the boot phase, it is
+     sufficient for the watchdog to enforce that each period being
+     programmed shall not be larger than the previous one.
+
+     The typical approach would be thus to configure through the pins a
+     timeframe reasonably long to allow for kernel boot and spawning of the
+     designated safe userspace process, which would then proceed with
+     lowering the ping threshold according to its needs.
+
+     Additional restrictions peppered along the booth sequence are possible,
+     but not strictly required.
+
+  * **Safety Island**
+
+    "Safety Island" is a fancy name for referring to what is typically a
+    special core, within a System on a Chip, that can be used as foundation
+    for building safety claims. Characteristics are:
+    *  Its HW design, implementation and validation are compatible with safety
+       requirements.
+    *  Usually is a microcontroller core
+    *  Its operating parameters cannot be affected by other components in
+       the system (e.g. the cores running Linux).
+    *  It can either directly or indirectly affect the operating parameters
+       of other components in the system (e.g. it can initiate a reset).
+    *  Usually runs a safe RTOS that has a safety qualification compatible
+       with the safety requirements for the overall system.
+    *  Can act as complex, programmable, watchdog.
+    *  Could even perform some simple safety related tasks, where it makes
+       sense.
+
+    Unsurprisingly, this sort of feature is not available in any randomly
+    picked SoC designs, however one could still implement a sort of
+     "poor man's safety island" by selecting a standalone component with
+    similar characteristics (e.g. a safety rated microcontroller w/ PCB)
+    and integrate it with the larger non-safety qualified system, through
+    some shared interface (e.g. CAN, I2C or SPI) giving said microcontroller
+    the ability to e.g. powercycle the main system, while also performing
+    potentially sophisticated monitoring, that can evolve through complex
+    state graphs, in lockstep with the primary system.
+ 
+    
+    This approach can be interesting as viable solution for converting
+    an existing non-safe system design into a safety qualifiable one.
+
+  * **Other Exception Context**
+ 
+    As discussed in the "Checklist" document, on a modern SoC it is not
+    uncommon to find out that the kernel is not the most privileged
+    execution context, and that other contexts are capable of interfering
+    with it in undetectable (by the kernel) ways. And the only solution
+    is likely to be requiring that these contexts must be safety qualified.
+    The hypervisor is a good example of such scenario.
```

Comment:
Maybe for completeness I would also mention the Trusted Execution Environment...